opensrp-opd/src/main/java/org/smartregister/opd/fragment/ProfileOverviewFragment.java

```diff
@@ -0,0 +1,170 @@
+import org.smartregister.opd.pojos.OpdCheckIn;
+import org.smartregister.opd.pojos.OpdVisit;
+import org.smartregister.opd.utils.FilePath;
+import org.smartregister.opd.utils.OpdConstants;
+import org.smartregister.opd.utils.OpdDbConstants;
+import org.smartregister.util.DateUtil;
+import org.smartregister.view.fragment.BaseProfileFragment;
+
+import java.io.IOException;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+import java.util.Locale;
+
+import timber.log.Timber;
+
+
+/**
+ * Created by Ephraim Kigamba - ekigamba@ona.io on 2019-09-30
+ */
+public class ProfileOverviewFragment extends BaseProfileFragment {
+
+    private List<YamlConfigWrapper> yamlConfigListGlobal;
+    private String baseEntityId;
+
+    public static ProfileOverviewFragment newInstance(Bundle bundle) {
+        Bundle args = bundle;
+        ProfileOverviewFragment fragment = new ProfileOverviewFragment();
+        if (args == null) {
+            args = new Bundle();
+        }
+        fragment.setArguments(args);
+        return fragment;
+    }
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+    }
+
+    @Override
+    protected void onCreation() {
+        if (getActivity() != null) {
+            yamlConfigListGlobal = new ArrayList<>();
+            baseEntityId = getActivity().getIntent().getStringExtra(OpdConstants.IntentKey.BASE_ENTITY_ID);
+        }
+    }
+
+    @Override
+    protected void onResumption() {
+        try {
+            yamlConfigListGlobal = new ArrayList<>(); //This makes sure no data duplication happens
+            Facts facts = new Facts();
+
+            // TODO: Move this to a background thread once Benn has added his work with the AppExecutors
+            OpdVisit visit = OpdLibrary.getInstance().getVisitRepository().getLatestVisit(baseEntityId);
+
+            if (visit != null) {
+                OpdCheckIn checkIn = OpdLibrary.getInstance().getCheckInRepository().getCheckInByVisit(visit.getId());
+
+                if (checkIn != null) {
+                    setDataFromCheckIn(checkIn, visit, facts);
+                }
+            }
+
+            generateYamlConfigList(facts);
+
+            if (getActivity() != null) {
+                ProfileOverviewAdapter adapter = new ProfileOverviewAdapter(getActivity(), yamlConfigListGlobal, facts);
+                adapter.notifyDataSetChanged();
+                // set up the RecyclerView
+                RecyclerView recyclerView = getActivity().findViewById(R.id.profile_overview_recycler);
+                recyclerView.setLayoutManager(new LinearLayoutManager(getActivity()));
+                recyclerView.setAdapter(adapter);
+            }
+
+        } catch (IOException e) {
+            Timber.e(e);
+        }
+    }
+
+    private void generateYamlConfigList(@NonNull Facts facts) throws IOException {
```

Comment:
Methods like this can belong in a presenter, or what got you think? also `setDataFromCheckIn`

Reply:
Yeap, moved to a presenter